Initial adjustments post rebase

Timo614 opened the pull request:
Some minor adjustments / new methods for https://github.com/big-neon/bn-api/pull/2.

Putting up a PR now as it's referenced by that branch but still wish to flesh this out a bit more with additional methods, etc. prior to merging.

mikethetike commented on src/models/artists.rs:
```diff
@@ -36,4 +36,33 @@ impl Artist {
+    pub fn all(conn: &Connectable) -> Vec<Artist> {
+        artists::table
+            .load(conn.get_connection())
+            .expect("Unable to load artists")
```
Can we use `DatabaseError::wrap()` here instead of `expect`

Timo614 commented:
I've adjusted this as per the review feedback to use the DatabaseError structs now for the db responses. I've also added a new API branch to go along with this one as the interface has changed now: https://github.com/big-neon/bn-api/pull/3

CjS77 commented on tests/unit/artists.rs:
```diff
@@ -1,11 +1,68 @@
+}
+
+#[test]
+fn all() {
```
Should add a test that checks that `all` works for n>1 artists too

Timo614 replied:
Sounds good, added it to the test.
